### ISSUE TITLE
Breaks web builds

### ISSUE THREAD

**Issue by krille-chan** [canonical/desktop_notifications.dart#5]
I'm creating an app which should run on mobile, desktop and web. When I add this package the app no longer builds the web version. Looks like there is a platform stuff missing.

```
[   +1 ms] Initializing file store
[   +7 ms] web_entrypoint: Starting due to {InvalidatedReason.inputChanged}
[   +9 ms] web_entrypoint: Complete
[   +3 ms] gen_localizations: Starting due to {}
[ +353 ms] "ar": 3 untranslated message(s).
[        ] "cs": 4 untranslated message(s).
[        ] "es": 11 untranslated message(s).
[        ] "eu": 11 untranslated message(s).
[        ] "hu": 25 untranslated message(s).
[        ] "hy": 25 untranslated message(s).
[        ] "it": 233 untranslated message(s).
[        ] "ja": 18 untranslated message(s).
[        ] "pl": 67 untranslated message(s).
[        ] "sk": 25 untranslated message(s).
[        ] "uk": 23 untranslated message(s).
[        ] To see a detailed report, use the --untranslated-messages-file 
           option in the tool to generate a JSON format file containing 
           all messages that need to be translated.
[   +1 ms] gen_localizations: Complete
[  +34 ms] dart2js: Starting due to {}
[+28555 ms] Persisting file store
[   +1 ms] Done persisting file store
[   +1 ms] Target dart2js failed: Exception: /opt/flutter/flutter/.pub-cache/hosted/pub.dartlang.org/dbus-0.0.0-dev.21/lib/src/getuid.dart:1:8:
           Error: Not found: 'dart:ffi'
           import 'dart:ffi';
                  ^
           /opt/flutter/flutter/.pub-cache/hosted/pub.dartlang.org/dbus-0.0.0-dev.21/lib/src/getuid.dart:3:20:
           Error: Type 'Int32' not found.
           typedef _getuidC = Int32 Function();
                              ^^^^^
           /opt/flutter/flutter/.pub-cache/hosted/pub.dartlang.org/dbus-0.0.0-dev.21/lib/src/getuid.dart:8:17:
           Error: Getter not found: 'DynamicLibrary'.
             final dylib = DynamicLibrary.open('libc.so.6');
                           ^^^^^^^^^^^^^^
           Error: Compilation failed.
[   +2 ms] Compiling lib/main.dart for the Web... (completed in 29.0s)
[   +1 ms] "flutter web" took 33,767ms.
[   +4 ms] Exception: Failed to compile application for the Web.
[        ] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
           #1      buildWeb (package:flutter_tools/src/web/compile.dart:89:5)
           #2      _rootRunUnary (dart:async/zone.dart:1198:47)
           #3      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #4      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #5      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #6      Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #7      Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #8      _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
           #9      _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
           #10     FlutterBuildSystem.build (package:flutter_tools/src/build_system/build_system.dart)
           #11     _rootRunUnary (dart:async/zone.dart:1198:47)
           #12     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #13     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #14     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #15     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #16     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #17     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
           #18     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
           #19     _BuildInstance.invokeTarget (package:flutter_tools/src/build_system/build_system.dart)
           #20     _rootRunUnary (dart:async/zone.dart:1198:47)
           #21     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #22     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #23     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #24     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #25     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #26     Future.wait. (dart:async/future.dart:414:23)
           #27     _rootRunUnary (dart:async/zone.dart:1198:47)
           #28     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #29     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #30     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #31     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #32     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #33     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
           #34     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
           #35     _BuildInstance.invokeTarget (package:flutter_tools/src/build_system/build_system.dart)
           #36     _rootRunUnary (dart:async/zone.dart:1198:47)
           #37     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #38     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #39     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #40     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #41     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #42     Future.wait. (dart:async/future.dart:414:23)
           #43     _rootRunUnary (dart:async/zone.dart:1198:47)
           #44     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #45     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #46     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #47     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #48     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #49     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
           #50     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
           #51     _BuildInstance._invokeInternal (package:flutter_tools/src/build_system/build_system.dart)
           #52     _asyncErrorWrapperHelper.errorCallback (dart:async-patch/async_patch.dart:91:64)
           #53     _rootRunBinary (dart:async/zone.dart:1214:47)
           #54     _CustomZone.runBinary (dart:async/zone.dart:1107:19)
           #55     _FutureListener.handleError (dart:async/future_impl.dart:157:20)
           #56     Future._propagateToListeners.handleError (dart:async/future_impl.dart:708:47)
           #57     Future._propagateToListeners (dart:async/future_impl.dart:729:24)
           #58     Future._completeError (dart:async/future_impl.dart:537:5)
           #59     _AsyncAwaitCompleter.completeError (dart:async-patch/async_patch.dart:47:15)
           #60     Dart2JSTarget.build (package:flutter_tools/src/build_system/targets/web.dart)
           #61     _rootRunUnary (dart:async/zone.dart:1198:47)
           #62     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #63     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #64     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #65     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #66     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #67     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
           #68     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
           #69     _run (package:flutter_tools/src/base/error_handling_io.dart)
           #70     _rootRunUnary (dart:async/zone.dart:1198:47)
           #71     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #72     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #73     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #74     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #75     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #76     Future.wait. (dart:async/future.dart:414:23)
           #77     _rootRunUnary (dart:async/zone.dart:1198:47)
           #78     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #79     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #80     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #81     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #82     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #83     Future._asyncCompleteWithValue. (dart:async/future_impl.dart:567:7)
           #84     _rootRun (dart:async/zone.dart:1190:13)
           #85     _CustomZone.run (dart:async/zone.dart:1093:19)
           #86     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
           #87     _CustomZone.bindCallbackGuarded. (dart:async/zone.dart:1037:23)
           #88     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
           #89     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
           #90     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
           #91     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)
[  +20 ms] ensureAnalyticsSent: 18ms
[   +2 ms] Running shutdown hooks
[   +3 ms] Shutdown hooks complete
[        ] exiting with code 1
```

**Comment by robert-ancell** [comment]
This should be opened against https://github.com/canonical/dbus.dart, but thanks for finding it!

**Comment by robert-ancell** [comment]
This is now fixed in dbus 0.3.2 - https://github.com/canonical/dbus.dart/pull/189

**Comment by robert-ancell** [comment]
Fixed in desktop_notifications 0.3.0
